### PR TITLE
Adding unit tests for the cache.

### PR DESCRIPTION
**Pull request by vishh**
Fixed a few bugs in the cache.

### CODE REVIEW

**Comment by vishh** [comment]
Ping @vmarmol

File: sinks/cache/gc_store.go [hunk 1]
```diff
@@ -43,13 +44,15 @@ func (gcs *gcStore) Delete(start, end time.Time) error {
 
 func (gcs *gcStore) reapOldData(timestamp time.Time) {
 	end := timestamp.Add(-gcs.bufferDuration)
-	start := time.Time{}
-	if err := gcs.store.Delete(start, end); err != nil {
-		glog.Fatalf("failed to delete old data - %v", err)
+	if end.Before(time.Unix(0, 0)) {
```
Comment (vmarmol) [inline sinks/cache/gc_store.go]: nit: `time.Time{}` or was there an issue with that?

**Comment by vmarmol** [comment]
LGTM, I thought some of those changes had been made for some reason.

File: sinks/cache/gc_store.go (hunk 1 above)
Comment (vishh) [inline sinks/cache/gc_store.go]: I don't think I intended to make this change. Merge issue maybe? Will fix it anyways.

Comment (vishh) [inline sinks/cache/gc_store.go]: Done.

**Comment by vishh** [comment]
PTAL.

**Comment by vmarmol** [comment]
LGTM, thanks @vishh!

**Pull request merged**